src/app/modules/core/services/validator.service.ts

```diff
@@ -0,0 +1,76 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+
+import { zip, Observable, of } from 'rxjs';
+import { switchMap, mergeMap, concatAll, toArray } from 'rxjs/operators';
+
+import range from 'src/app/modules/core/utils/range';
+import { BeaconNodeService } from './beacon-node.service';
+import { WalletService } from './wallet.service';
+
+import {
+  ValidatorBalances,
+} from 'src/app/proto/eth/v1alpha1/beacon_chain';
+
+export const MAX_EPOCH_LOOKBACK = 5;
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ValidatorService {
+  constructor(
+    private http: HttpClient,
+    private beaconNodeService: BeaconNodeService,
+    private walletService: WalletService,
+  ) { }
+
+  // Observables.
+  recentEpochBalances(currentEpoch: number, lookback: number): Observable<ValidatorBalances[]> {
+    if (lookback > MAX_EPOCH_LOOKBACK) {
+      throw new Error('Cannot request greater than max lookback epochs');
```

Comment:
If this will be displayed to the user, then they should know what is the maximum.